app/services/reporter.rb

```diff
@@ -6,61 +6,80 @@
 # run queries and produce reports from the results, for consumption
 # by preservation catalog maintainers
 class Reporter
-  attr_reader :storage_root, :druids
+  attr_reader :storage_root
 
   # @params [Hash] params used to initialize the Reporter service
   # @return [Reporter] the reporter
   def initialize(params)
     @storage_root = MoabStorageRoot.find_by!(name: params[:storage_root_name])
-    @druids = moab_storage_root_druid_list
+    @msr_names = {}
   end
 
-  # @return [Array] an array of druids on the storage root
-  def moab_storage_root_druid_list
-    druid_array = []
+  def moab_storage_root_list_pres_obj_comp_moabs_relation
     PreservedObject
       .joins(:complete_moabs)
       .where(complete_moabs: { moab_storage_root: storage_root })
-      .select(:druid)
       .order(:druid)
+  end
+
+  # @return [Array] an array of druids on the storage root
+  def druid_csv_list
+    druid_array = [['druid']]
+    moab_storage_root_list_pres_obj_comp_moabs_relation
+      .select(:druid)
```

Comment:
Is there any chance we will get duplicate druids in this list? And, if we do, do we care about druids being in the CSV list multiple times?

Reply:
it is possible in that the schema allows for multiple `CompleteMoab`s for a given `PreservedObject`.  in practice, we never got around to supporting this in the rest of the code, and some of our shameless green implementations elsewhere assume one `CompleteMoab` per `PreservedObject`, iirc.  also, in practice, i don't think we'd put the same moab on the same storage root multiple times.  so i think that in practice, the druids will be unique in this list, both because druids are currently unique across `CompleteMoab`s, and because even in our more robust future, they'd likely still be unique for all `CompleteMoab`s on a given storage root.

Reply:
As long as you think this is unlikely to cause heartache or confusion, this is all good. Otherwise, we could do a distinct query but that would open up the can of worms of which row to  prefer over the others.

Reply:
we could add a comment stating `there will only be uniq druid values for a single storage root`